File: 프로그래머스/0/181934. 조건 문자열/조건 문자열.java
class Solution {
    public int solution(String ineq, String eq, int n, int m) {
    
        int result;
        
        if(ineq.equals(">")&& eq.equals("=")){
            result = n>=m ? 1 : 0;
        }else if(ineq.equals("<")&& eq.equals("=")){
            result = n<=m ? 1 : 0;
        }else if(ineq.equals(">")&& eq.equals("!")){
            result = n>m ? 1 : 0;
        }else{
            result = n<m ? 1 : 0;
        }
        
        return result;
    }
}

File: 프로그래머스/0/181934. 조건 문자열/조건 문자열.kt
class Solution {
    fun solution(ineq: String, eq: String, n: Int, m: Int): Int {
        var answer: Int = 0
        var isRight: Boolean=false
        var condition : String =""
        condition=ineq+eq
        when(condition){
            ">=" -> isRight = n>=m
            "<=" -> isRight = n<=m
            ">!" -> isRight = n>m
            "<!" -> isRight = n<m
        }
        
        if(isRight){
            answer=1
        }
        return answer
    }
}